src/main/java/com/keniu/services/BookServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.keniu.services;
+
+import com.keniu.dtos.BookDto;
+import com.keniu.dtos.CreateBookRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.BookMapper;
+import com.keniu.repositories.BookRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(createBookRequestDto)));
```

Comment:
it's better to create local variable for Book `bookMapper.toModel(createBookRequestDto)`

Reply:
For what?

Reply:
to improve readability and comfortable debugging
```suggestion
        Book book = bookMapper.toModel(createBookRequestDto);
        return bookMapper.toDto(bookRepository.save(book));
```

Reply:
I don't set any value in it.
I only use it once.
The variable’s value carries no real meaning and doesn’t clarify anything to the reader.
In short, using the variable here is redundant.

Reply:
it's only up to you, write readable code (not only for you but for other developers)

Reply:
Code is more readable when we avoid creating unnecessary variables.

Reply:
Matter of personal preference/project conventions